src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/CostFunctionsForPhysicalPlansImpl.java

```diff
@@ -0,0 +1,121 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.utils.CardinalityEstimation;
+
+public class CostFunctionsForPhysicalPlansImpl implements CostFunctionsForPhysicalPlans{
+    protected final CostFunctionsForRootOperators costFunctionForRoot;
+    // protected final PhysicalPlanCostCache physicalPlanCostCache;
```

Comment:
You don't need this cache here. The purpose of this class can be to provide only the actual cost functions, and the option of caching the costs of plans can be handled completely outside of this class, exactly as it is currently done already in the 'CostModel' class. Hence, you can safely remove all these portions of commented code from this class.